src/components/organisms/PaneManager/PaneManagerSplitView.tsx

```diff
@@ -25,139 +20,61 @@ const TemplateManagerPane = React.lazy(() => import('@organisms/TemplateManagerP
-    }
-
-    if (!leftActive && navPaneWidth && splitViewContainerWidth) {
-      dispatch(
-        setPaneConfiguration({
-          ...paneConfiguration,
-          navWidth: navPaneWidth / (splitViewContainerWidth || 1.0),
-        })
-      );
-    }
-
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [leftActive]);
+  const handleResize = useCallback(
+    elements => {
+      const updates = elements.reduce((obj: any, el: any) => {
+        if (!['leftPane', 'navPane'].includes(el.props.id)) return obj;
+        obj[el.props['id']] = el.props['flex'];
+        return obj;
+      }, {});
+
+      const newLayout = {...layout, ...updates};
+      dispatch(setPaneConfiguration(newLayout));
+    },
+    [dispatch, layout]
+  );
 
   return (
     <S.SplitViewContainer>
       <ValidationPane height={paneHeight} />
 
-      <S.SplitViewGrid ref={splitViewContainerRef as any} $gridTemplateColumns={splitViewGridTemplateColumns}>
+      <ReflexContainer orientation="vertical" onStopResize={handleResize} windowResizeAware>
         {leftActive && leftMenuSelection && (
-          <S.LeftPaneContainer ref={leftPaneRef}>
-            <ResizableBox
-              height={paneHeight}
-              width={splitViewContainerWidth * leftWidth}
-              minConstraints={[MIN_SPLIT_VIEW_PANE_WIDTH, paneHeight]}
-              maxConstraints={[leftPaneMaxWidth, paneHeight]}
-              axis="x"
-              resizeHandles={['e']}
-              handle={(h: number, ref: LegacyRef<HTMLSpanElement>) => (
-                <span className="custom-modal-handle" ref={ref} />
-              )}
-              onResizeStop={resizeLeftPane}
-            >
-              <S.Pane id="LeftPane">
-                <Suspense fallback={null}>
-                  {leftMenuSelection === 'file-explorer' && <FileTreePane />}
-                  {leftMenuSelection === 'helm-pane' && <HelmPane />}
-                  {leftMenuSelection === 'kustomize-pane' && <KustomizePane />}
-                  {leftMenuSelection === 'templates-pane' && <TemplateManagerPane contentHeight={paneHeight} />}
-                </Suspense>
-              </S.Pane>
-            </ResizableBox>
-          </S.LeftPaneContainer>
+          <ReflexElement id="leftPane" minSize={MIN_SPLIT_VIEW_PANE_WIDTH} flex={layout.leftPane}>
+            <S.LeftPane>
+              <Suspense fallback={<div />}>
+                {leftMenuSelection === 'file-explorer' && <FileTreePane />}
+                {leftMenuSelection === 'helm-pane' && <HelmPane />}
+                {leftMenuSelection === 'kustomize-pane' && <KustomizePane />}
+                {leftMenuSelection === 'templates-pane' && <TemplateManagerPane contentHeight={paneHeight} />}
+              </Suspense>
+            </S.LeftPane>
+          </ReflexElement>
         )}
 
-        <S.Pane id="NavPane" $height={paneHeight} ref={navPaneRef}>
+        {leftActive && leftMenuSelection && <ReflexSplitter propagate />}
```

Comment:
📓 just to spread some awareness: You can't use fragments within ReflexElement so you need to do this separately instead of grouping these within a fragment.

Reply:
I would add this as a comment next to the code because it's useful to know